Parse HTML data in a webview

stuartlangridge opened the pull request:
Allow passing a URL on the command line (`qmlscene app/Main.qml http://mikemai.net/demos/ios-web-app.html` is dead handy, and we don't have to parse the arguments properly because we just look for the one which is a URL!).

Load the requested URL in an invisible webview and parse data out of it to get title and icon.

We need to actually fetch the icon itself, not just display the remote one, and at the moment we just use the first icon we find (and probably blow up if there isn't one) rather than correctly choosing the one we want. But, progress.

(Work on issue #2)

stuartlangridge committed:
Allow passing a URL on the command line. Load the requested URL in an invisible webview and parse data out of it to get title and icon.

rschroll commented:
Yeah, this breaks if it can't find an icon.  There are several places 
where this could be solved; I don't know which is best offhand.  It's 
probably a good idea to check for favicons and opengraph images as 
well, but the only think I want right now is for it to not break.

Can we try to keep lines under ~100 characters?  I'm sure I'm breaking 
this rule some places already, but I like to avoid horizontal scrolling 
in my editor if possible.  Two places jump out at me: line 216 of 
Main.qml and line 11 of manifestParser.js.  This latter has so much 
going on that it's nearly impossible to read.  Let's break the 
querySelector out into its own statement.

Speaking of pedantry, I like the "short summary (< 72 chars) followed 
by the longer paragraph (if needed) description" format for commit 
messages.  I wouldn't reject a commit over this, but I might edit the 
message myself....

stuartlangridge committed:
Icon choice and reformatting changes
Look through the list of returned icons and choose the best by size, and have a fallback if no icon is found.
Reformat some lines to fit under ~100 characters.

stuartlangridge commented:
All good points, and pushed changes to meet them.

rschroll commented:
Looks good.  Thanks!

Out of curiosity, is there a reason you used a functional approach 
(icons.forEach()) rather than a procedural approach (for loop) in going 
through the available icons?  Is there a performance benefit, or is it 
just your style.  (I would never have considered not using a for loop.)

stuartlangridge commented:
Mostly style, if I'm honest. I tend to do functionalish things because `.map()` and `.filter()` are useful, and occasionally it's good that `forEach` creates a closure where a `for` loop doesn't, but it's mostly just habit rather than anything I'd justify in strong technical terms.
